### ISSUE TITLE
Prepare for Release 1.0.0 of ACA-Py

### ISSUE THREAD

**Issue by swcurran** [hyperledger/aries-cloudagent-python#2753]
The following is a checklist of items to be completed for Release 1.0.0 of ACA-Py:

- [x] Flag PRs for 1.0.0 (may be merged into 0.12.0)
- [x] Complete PRs flagged for 1.0.0
- [x] Flag Issues for 1.0.0
- [x] Complete Issues flagged for 1.0.0
- [x] Review/discuss/define LTS for ACA-Py, as per [RFC 0799 Long Term Support](https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0799-long-term-support)
- [x] From Supported RFCs list as "not completed" for AIP 2.0 `RFC 0587 -- Support for the DIDComm V2 envelope format is a work in progress`
- [x] Update AIP 2.0 to remove `Please Ack`
- [ ] Blog Post/Press Release

**Comment by ff137** [comment]
I'd like to propose a few suggestions to consider for the 1.0.0 release:

- Upgrade Python to 3.12 (I've tested locally and things pass smoothly)
- review TODO comments in the codebase ... ideally resolve all of them! But primarily just find all and track important ones.
- In PR tests, there are 296 skipped, and 6 xfailed. Perhaps the skipped tests are worth reviewing and cleaning up where applicable, and perhaps a reinspection of the expected failures too.
- There is a ruff lint rule that's currently being ignored: `D417`. This rule is for missing arguments in the description of docstrings. 
  - This is possibly the most minor of "nice to haves", but  there are _a lot_ of methods (281) with missing descriptions, and I feel that this is important enough for maintainability to mention. In an ideal world these can be solved too before a major release. 
  - Will require a very patient contributor, possibly with help from GPT / co-pilot to automate some docstrings

Just some thoughts that came to mind! It's very much "nice to have" territory, but probably worth considering some of these before wanting to go maintenance mode.

**Comment by WadeBarnes** [comment]
I like the idea of upgrading the Python version.

**Comment by swcurran** [comment]
Thanks @ff137 — @dbluhm @jamshale @ianco and other maintainers, please weigh in on this either/both here and at the next ACA-Pug meeting.

**Comment by jamshale** [comment]
I'll be taking this over and trying to push the release forward. I like @ff137's suggestions and will be upgrading python and going over todo's and fixme's.

**Comment by jamshale** [comment]
Also, I think we should make sure the AATH passes with the new credo backchannel updates.

**Comment by swcurran** [comment]
That is correct — 1.0.0 has not yet been named an LTS.  We are expecting a 1.0.1 very shortly, and perhaps then we’ll make it an LTS.  At that time, the references you indicate will be updated.  We probably can just go ahead with that, but we’re still getting used to the LTS process so waiting on a bit more feedback from the community :-).

**Comment by swcurran** [comment]
Yes, the LTS in this scenario would be “1.0”, with the branch occuring at the 1.0.1 release. The issue is at what point we declare the “LTS”. The maintainers agreed that doing so as a new major release occured would be a bad idea, in case an immediate breaking change occured requiring a 1.1.0 release after LTS was declared for 1.0.  So we will wait a bit before the declaration.

Does that description still match your expectations?

**Comment by rngadam** [comment]
> Yes, the LTS in this scenario would be “1.0”, with the branch occuring at the 1.0.1 release. The issue is at what point we declare the “LTS”. The maintainers agreed that doing so as a new major release occured would be a bad idea, in case an immediate breaking change occured requiring a 1.1.0 release after LTS was declared for 1.0. So we will wait a bit before the declaration.
> 
> Does that description still match your expectations?

Ah, ok! Yes, I understand now. Waiting for 1.0.1 is just giving an opportunity to find a major issue. If found, we restart the waiting period for the next MINOR and then PATCH bump.
